recipes/osgearth/all/conanfile.py

```diff
@@ -65,150 +93,202 @@ class OsgearthConan(ConanFile):
+        # in a shared build. osgearth assumes that these headers are available, however. What this means is that we
+        # cannot currently support shared builds of rocksdb.
+        if self.options.build_rocksdb_cache:
+            is_rocksdb_shared = self.dependencies["rocksdb"].options.shared
+
+            if is_rocksdb_shared:
+                raise ConanInvalidConfiguration('The "build_rocksdb_cache" option cannot be used with a shared build of rocksdb.')
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
     def config_options(self):
         if self.settings.os != "Windows":
             self.options.enable_wininet_for_http = False
 
         if self.settings.os == "Windows":
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
-        if self.settings.compiler == "Visual Studio":
+        if is_msvc(self) and Version(self.version) < "3.7.2":
             self.options.build_procedural_nodekit = False
 
-        if self.settings.compiler == "gcc" and self.settings.compiler.version == "11":
-            # need draco >= 1.4.0 for gcc11
-            # https://github.com/google/draco/issues/635
-            self.options.with_draco = False
+        if Version(self.version) < "3.7.2":
+            if self.settings.compiler == "gcc" and self.settings.compiler.version == "11":
+                # need draco >= 1.4.0 for gcc11
+                # https://github.com/google/draco/issues/635
+                self.options.rm_safe("with_draco")
+        else:
+            self.options.rm_safe("build_leveldb_cache")
+            self.options.rm_safe("with_draco")
 
     def requirements(self):
-
         self.requires("opengl/system")
         self.requires("gdal/3.4.3")
         self.requires("openscenegraph/3.6.5")
-        self.requires("libcurl/8.2.0")
+        self.requires("libcurl/8.12.1")
         self.requires("lerc/2.2")
         self.requires("rapidjson/1.1.0")
 
-        self.requires("zlib/[>=1.2.11 <2]")  # override
-        self.requires("libtiff/4.5.1")  # override
-        self.requires("libpng/1.6.40")  # override
-
         # if self.options.build_triton_nodekit:
         #     self.requires("triton_nodekit")
         # if self.options.build_silverlining_nodekit:
         #     self.requires("silverlining_nodekit")
-        if self.options.build_leveldb_cache:
+
+        if self.options.get_safe("build_leveldb_cache"):
             self.requires("leveldb/1.22")
         if self.options.build_rocksdb_cache:
             self.requires("rocksdb/6.29.5")
         if self.options.build_zip_plugin:
-            self.requires("zstd/1.5.5")  # override
+            self.requires("libzip/1.7.3")
         if self.options.with_geos:
             self.requires("geos/3.11.1")
         if self.options.with_sqlite3:
             self.requires("sqlite3/[>=3.42 <4]")
```

Comment:
The [Dependencies documentation](https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges) would indicate that this version range specification for `sqlite3` is disallowed. Yet, it predates the changes in this PR...

Reply:
Okay, then I'll go ahead and mark this comment as resolved. Thanks!